event/monitoring.go

```diff
@@ -66,6 +66,10 @@ type CommandMonitor struct {
 	Failed    func(context.Context, *CommandFailedEvent)
 }
 
+type CommandInterceptor struct {
+	Process func(context.Context, *[]byte) error
```

Comment:
`*[]byte` is definitely not the right thing here. I'm not sure if it should be a `bson.Raw`, or an `Operation`, or something else entirely. Things I would want to consider circuit breaking on:
- Cluster
- Database
- Collection
- Operation type (reads vs writes)
- Specific query (an expensive aggregation across all users should be shut off but allow individual user lookups)